src/AutoRest.CSharp/LowLevel/Output/LowLevelClient.cs

```diff
@@ -156,13 +156,19 @@ private IEnumerable<ConstructorSignature> BuildPrimaryConstructors(IReadOnlyList
 
             if (Fields.CredentialFields.Count == 0)
             {
-                yield return CreatePrimaryConstructor(requiredParameters.Concat(optionalToRequired).ToArray());
+                /* order the constructor paramters as (endpoint, requiredParameters, OptionalParamters).
+                 * use the OrderBy to put endpoint as the first parameter.
+                 * */
+                yield return CreatePrimaryConstructor(requiredParameters.Concat(optionalToRequired).OrderBy(parameter => parameter.Name != "endpoint").ToArray());
```

Comment:
Do we need to care about the case when comparing 'endpoint'?

Reply:
`endpoint` is current hardcoded. but it is better to compare case-insensitive. updated. Thanks,